service/src/main/kotlin/app/cash/backfila/service/runner/statemachine/BatchPrecomputer.kt

```diff
@@ -69,6 +71,7 @@ class BatchPrecomputer(
           backfillRunner.factory.transacter.transaction { session ->
             val dbRunPartition = session.load(backfillRunner.partitionId)
             dbRunPartition.precomputing_done = true
+            saveProgress(dbRunPartition)
 
```

Comment:
Do we want a precomputer completed event?

Reply:
there is, just below

Reply:
There is a stopped? but that doesn't mean it is complete? On second thought maybe it does.

Reply:
it does.
 right below:
```              DbEventLog(
                backfillRunner.backfillRunId,
                partition_id = backfillRunner.partitionId,
                type = DbEventLog.Type.STATE_CHANGE,
                message = "precomputing complete"
```